pkg/apis/certmanager/v1/types_certificaterequest.go

```diff
@@ -48,63 +50,95 @@ const (
 // configured issuers.
 //
 // All fields within the CertificateRequest's `spec` are immutable after creation.
-// A CertificateRequest will either succeed or fail, as denoted by its `status.state`
-// field.
+// A CertificateRequest will either succeed or fail, as denoted by its `Ready` status
+// condition and its `status.failureTime` field.
 //
 // A CertificateRequest is a one-shot resource, meaning it represents a single
 // point in time request for a certificate and cannot be re-used.
 // +k8s:openapi-gen=true
 type CertificateRequest struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// Desired state of the CertificateRequest resource.
+	// Specification of the desired state of the CertificateRequest resource.
+	// https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
 	Spec CertificateRequestSpec `json:"spec"`
 
-	// Status of the CertificateRequest. This is set and managed automatically.
+	// Status of the CertificateRequest.
+	// This is set and managed automatically.
+	// Read-only.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
 	// +optional
 	Status CertificateRequestStatus `json:"status"`
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// CertificateRequestList is a list of Certificates
+// CertificateRequestList is a list of CertificateRequests.
 type CertificateRequestList struct {
 	metav1.TypeMeta `json:",inline"`
-	metav1.ListMeta `json:"metadata"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
 
+	// List of CertificateRequests
 	Items []CertificateRequest `json:"items"`
 }
 
 // CertificateRequestSpec defines the desired state of CertificateRequest
+//
+// NOTE: It is important to note that the issuer can choose to ignore and/
+// or change any of the requested attributes. How the issuer maps a certificate
```

Comment:
**nit:** To keep things readable, I propose to avoid "and/or" unless necessary. Here, it doesn't matter if the issuer ignores and changes (conjunction) or if the issuer ignores or changes (disjunction). What's important to the reader is that the issuer has either changed or ignored the attribute.

```suggestion
// NOTE: It is important to note that the issuer can choose to ignore or
// change any of the requested attributes. How the issuer maps a certificate
```